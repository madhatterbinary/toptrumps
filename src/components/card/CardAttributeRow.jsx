import React from 'react';
import PropTypes from 'prop-types';
import { STATUS_READY, STATUS_DONE, STATUS_OVER } from '../../constants/constants';

const CardAttributeRow =(props) => {
    const {
      attributeKey,
      label,
      value,
      status,
      selectedAttribute,
      handleAttributeClick
    } = props;

  const handleAttributeSelected = () => {
    handleAttributeClick(attributeKey);
  }
  console.log("the status ", status);
  const attributeValue = status === STATUS_READY ? (
    <div
      className={`card-attribute-value ${selectedAttribute === attributeKey && 'selected'}`}
      onClick={handleAttributeSelected}
    >{value}</div>
  ) : (
    <span
      className={`card-attribute-value ${selectedAttribute === attributeKey && 'selected'}`}
    >{value}</span>);

  return (
    <div 
    className={status === STATUS_DONE || status === STATUS_OVER ? "card-attribute-row disabled" : "card-attribute-row"}
    onClick={handleAttributeSelected}
    >
      <span className={`card-attribute-label ${selectedAttribute === attributeKey && 'selected'}`}>
        {label}:
      </span>
      {attributeValue}
    </div>
  );
}

CardAttributeRow.propTypes = {
    attributeKey: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    handleAttributeClick: PropTypes.func.isRequired,
    status: PropTypes.string.isRequired
}

export default CardAttributeRow;
