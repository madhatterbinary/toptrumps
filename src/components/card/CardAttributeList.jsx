import React from 'react';
import PropTypes from 'prop-types';
import CardAttributeRow from './CardAttributeRow';
import attributes from '../../constants/attributes';

const CardAttributeList = (props) => {
  const {
    card,
    handleAttributeSelection,
    status,
    selectedAttribute,
  } = props;

  const attributeListItems = Object.keys(attributes).map((attributeKey) => {
    return (
    <CardAttributeRow
      key={`attribute-${attributeKey}`}
      attributeKey={attributeKey}
      label={attributes[attributeKey]}
      value={card[attributeKey] || ""}
      handleAttributeClick={handleAttributeSelection}
      selectedAttribute={selectedAttribute}
      status={status}
    />
    );
  });

  return (
    <div className="card-attribute-list">
      {attributeListItems}
    </div>
  );
}

CardAttributeList.propTypes = {
  card: PropTypes.object.isRequired,
  handleAttributeSelection: PropTypes.func.isRequired,
  status: PropTypes.string.isRequired,
  selectedAttribute: PropTypes.string
}

export default CardAttributeList;
