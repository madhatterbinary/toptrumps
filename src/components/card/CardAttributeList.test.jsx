import React from 'react';
import CardAttributeList from './CardAttributeList';
import CardAttributeRow from './CardAttributeRow';
import TestRenderer from 'react-test-renderer';

describe('CardAttributeList', () => {
  it('renders', () => {
    const component = TestRenderer.create(<CardAttributeList card={{}} handleAttributeSelection={() =>{}} status="" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
     const list = component.root.findAllByType(CardAttributeRow);
    expect(list.length).toEqual(6);
  });
});
