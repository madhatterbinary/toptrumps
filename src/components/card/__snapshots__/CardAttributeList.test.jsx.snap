// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardAttributeList renders 1`] = `
<div
  className="card-attribute-list"
>
  <div
    className="card-attribute-row"
    onClick={[Function]}
  >
    <span
      className="card-attribute-label false"
    >
      Cost
      :
    </span>
    <span
      className="card-attribute-value false"
    >
      
    </span>
  </div>
  <div
    className="card-attribute-row"
    onClick={[Function]}
  >
    <span
      className="card-attribute-label false"
    >
      Length
      :
    </span>
    <span
      className="card-attribute-value false"
    >
      
    </span>
  </div>
  <div
    className="card-attribute-row"
    onClick={[Function]}
  >
    <span
      className="card-attribute-label false"
    >
      Speed
      :
    </span>
    <span
      className="card-attribute-value false"
    >
      
    </span>
  </div>
  <div
    className="card-attribute-row"
    onClick={[Function]}
  >
    <span
      className="card-attribute-label false"
    >
      Crew
      :
    </span>
    <span
      className="card-attribute-value false"
    >
      
    </span>
  </div>
  <div
    className="card-attribute-row"
    onClick={[Function]}
  >
    <span
      className="card-attribute-label false"
    >
      Passengers
      :
    </span>
    <span
      className="card-attribute-value false"
    >
      
    </span>
  </div>
  <div
    className="card-attribute-row"
    onClick={[Function]}
  >
    <span
      className="card-attribute-label false"
    >
      Cargo Capacity
      :
    </span>
    <span
      className="card-attribute-value false"
    >
      
    </span>
  </div>
</div>
`;
